packages/zapp/console/src/components/WorkflowGraph/WorkflowGraph.tsx

```diff
@@ -21,54 +19,22 @@ export interface WorkflowGraphProps {
   onPhaseSelectionChanged: (phase: TaskExecutionPhase) => void;
   selectedPhase?: TaskExecutionPhase;
   isDetailsTabClosed: boolean;
-  workflow: Workflow;
-}
-
-interface PrepareDAGResult {
-  dag: dNode | null;
-  staticExecutionIdsMap?: any;
-  error?: Error;
 }
 
 const debug = createDebugLogger('@WorkflowGraph');
-
-function workflowToDag(workflow: Workflow): PrepareDAGResult {
-  try {
-    if (!workflow.closure) {
-      throw new Error('Workflow has no closure');
-    }
-    if (!workflow.closure.compiledWorkflow) {
-      throw new Error('Workflow closure missing a compiled workflow');
-    }
-    const { compiledWorkflow } = workflow.closure;
-    const { dag, staticExecutionIdsMap } = transformerWorkflowToDag(compiledWorkflow);
-
-    debug('workflowToDag:dag', dag);
-
-    return { dag, staticExecutionIdsMap };
-  } catch (e) {
-    return {
-      dag: null,
-      error: e as Error,
-    };
-  }
-}
-
 export interface DynamicWorkflowMapping {
   rootGraphNodeId: CompiledNode;
   dynamicWorkflow: any;
   dynamicExecutions: any[];
 }
 export const WorkflowGraph: React.FC<WorkflowGraphProps> = (props) => {
-  const {
-    onNodeSelectionChanged,
-    onPhaseSelectionChanged,
-    selectedPhase,
-    isDetailsTabClosed,
-    workflow,
-  } = props;
+  const { onNodeSelectionChanged, onPhaseSelectionChanged, selectedPhase, isDetailsTabClosed } =
+    props;
   const nodeExecutionsById = useContext(NodeExecutionsByIdContext);
-  const { dag, staticExecutionIdsMap, error } = workflowToDag(workflow);
+  const { compiledWorkflowClosure } = useNodeExecutionContext();
+  const { dag, staticExecutionIdsMap, error } = compiledWorkflowClosure
+    ? transformerWorkflowToDag(compiledWorkflowClosure)
+    : { dag: {}, staticExecutionIdsMap: {}, error: null };
```

Comment:
Few thoughts here:
1. I'm leaning towards passing optional `compileWorkflowClosure` to `transformerWorkflowToDag` and having early return with defaults there
2. Exact same ternary (only the defaults change based on the fields we're destructuring) is called in all 3 tabs. I'd suggest we call 'transformer' on the `ExecutionTab` level and pass `dag, staticExecutionIdsMap, error, ...` as individual props to specific components. Maybe we can treat errors on the tab level too.
3. Same with `dynamicParents` below, all 3 tabs now call the same query.

cc @jsonporter curious about your opinion on these points